service_container.rst

```diff
@@ -227,131 +209,182 @@ and set it on a ``$logger`` property::
         }
     }
 
+That's it! The container will *automatically* know to pass the ``logger`` service
+when instantiating the ``MessageGenerator``. How does it know to do this?
+:doc:`Autowiring </service_container/autowiring>`. The key is the ``LoggerInterface``
+type-hint in your ``__construct()`` method and the ``autowire: true`` config in
+``services.yml``. When you type-hint an argument, the container will automatically
+find the matching service. If it can't or there is any ambiguity, you'll see a clear
```

Comment:
there won't be ambiguities anymore for interfaces (interfaces can be autowired by aliasing only now, when not using the deprecated mode). Dunno how that applies here :)